File: milestone 2/series5/approxnorm.m
%Function generates n samples of a normally distibuted random variable. Two
%methods are used: a CLT approximation with 12 factors and the rand routine
%in MATLAB. The two methods are compared via qqplots and histograms
% Input:  n    number of samples
approxnorm1(100)

function[]=approxnorm1(n)
approx_1=sum(rand(n,12),2)-6;
approx_2=randn(n,1);
figure(1)
qqplot(approx_1)
title('CLT approximation')
figure(2)
qqplot(approx_2)
title('randn sample')
figure(3)
hist(approx_1,-5:0.1:5)
title('CLT approximation')
figure(4)
hist(approx_2,-5:0.1:5)
title('randn sample')
end

File: milestone 2/series5/approxnorm.py
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats

def approxnorm(n):
    approx_1 = np.sum(np.random.rand(n, 12), axis=1) - 6
    approx_2 = np.random.randn(n)

    plt.figure(1)
    stats.probplot(approx_1, dist="norm", plot=plt)
    plt.title('CLT approximation')

    plt.figure(2)
    stats.probplot(approx_2, dist="norm", plot=plt)
    plt.title('randn sample')

    plt.figure(3)
    plt.hist(approx_1, bins=np.arange(-5, 5, 0.1), alpha=1)
    plt.title('CLT approximation')

    plt.figure(4)
    plt.hist(approx_2, bins=np.arange(-5, 5, 0.1), alpha=1)
    plt.title('randn sample')

    plt.show()